From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mio <mio@mio19.uk>
Date: Tue, 20 Oct 2026 19:07:51 +1100
Subject: [PATCH] watch-power: adapt journald configuration to current nixpkgs

The npins bump to nixos-26.11pre1095516 removes the deprecated
services.journald.extraConfig option. Use the structured
services.journald.settings.Journal attribute set instead so the watch
power profile continues to configure volatile, size-limited journal
storage.

Signed-off-by: Mio <mio@mio19.uk>
---
 modules/watch-power.nix | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/modules/watch-power.nix b/modules/watch-power.nix
index 0432730a..5b936f61 100644
--- a/modules/watch-power.nix
+++ b/modules/watch-power.nix
@@ -39,10 +39,10 @@ in
 
     # Systemd journal volatile storage reduces write cycles on watch eMMC/UFS storage
     # and saves flash controller power draw.
-    services.journald.extraConfig = ''
-      Storage=volatile
-      RuntimeMaxUse=16M
-    '';
+    services.journald.settings.Journal = {
+      Storage = "volatile";
+      RuntimeMaxUse = "16M";
+    };
 
     # Sysctl kernel tuning for battery life and wrist devices
     boot.kernel.sysctl = {
